includes/data-events/class-webhooks.php

```diff
@@ -217,7 +224,33 @@ public static function create_endpoint( $url, $actions = [], $global = false ) {
 		}
 		\update_term_meta( $endpoint['term_id'], 'actions', $actions );
 		\update_term_meta( $endpoint['term_id'], 'global', $global );
-		return self::get_endpoint( $endpoint['term_id'] );
+		return self::get_endpoint_by_term( $endpoint['term_id'] );
+	}
+
+	/**
+	 * Register a webhook endpoint. For plugins use. See `newspack_webooks_register_endpoint` action.
+	 *
+	 * @param string $id      An unique identifier for the endpoint.
+	 * @param string $url     Endpoint URL.
+	 * @param array  $actions Array of action names.
+	 * @param bool   $global  Whether the endpoint should be triggered for all actions.
+	 *
+	 * @return void
+	 */
+	public static function register_system_endpoint( $id, $url, $actions = [], $global = false ) {
+
```

Comment:
We should add some validation here and return an error if the ID is already being used by an existing endpoint (system or term).

Reply:
The function will override the previous endpoint if you register a new one with the same ID, but this will only look at system endpoints...

I didn't want to have to query for all endpoints to do this check because system endpoints will be registered on ìnit` in every request (maybe we could be smarter about this and only but that's another topic)

What do you think of making sure that the ID is a string - and not a number that could conflict with endpoints stored in the DB?

As for the possibility of one plugin overriding the endpoint of another, that's the nature of plugins

Reply:
System endpoints could have their ID generated inside the registration method with a hardcoded prefix so they're at least conflict-free with term endpoints. Something like:

```php
public static function register_system_endpoint( $slug, $url, $actions = [], $global = false ) {
  $id = 'system-' . sanitize_title( $slug );
  ...
}
```

Reply:
> hardcoded prefix so they're at least conflict-free with term endpoints.

Don't like that because when you register an endpoint you'll want to use the ID to identify it later...

I added some validation in https://github.com/Automattic/newspack-plugin/pull/2446/commits/a6fe32b44c58001b20d3804adf346af560e2cd7f, see what you think

Reply:
Good call! I liked your approach better